libs/langgraph/src/prebuilt/react_agent_executor.ts

```diff
@@ -142,6 +143,103 @@ function _getPrompt(
   return _getPromptRunnable(finalPrompt);
 }
 
+function _shouldBindTools(
+  llm: LanguageModelLike,
+  tools: (StructuredToolInterface | DynamicTool | RunnableToolLike)[]
+): boolean {
+  if (!Runnable.isRunnable(llm) || !("kwargs" in llm)) {
+    return true;
+  }
+
+  if (
+    !llm.kwargs ||
+    typeof llm.kwargs !== "object" ||
+    !("tools" in llm.kwargs)
+  ) {
+    return true;
+  }
+
+  type OpenAITool = {
+    type: "function";
+    function: {
+      name: string;
+      description: string;
+      parameters: {
+        type: "object";
+        properties: Record<string, unknown>;
+        required: string[];
+        additionalProperties: boolean;
+        $schema: string;
+      };
+    };
+  };
+
+  type AnthropicTool = {
+    name: string;
+    description: string;
+    input_schema: Record<string, unknown>;
+  };
```

Comment:
these feel out of place here - can we import these from langchain/core?

Reply:
they don't exist there since they are library-specific

Reply:
actually figured out a way to simplify